test/scripts/write_fake_dicoms.py

```diff
@@ -0,0 +1,315 @@
+#!/usr/bin/env python3
+
+#  Copyright (c) University College London Hospitals NHS Foundation Trust
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import json
+
+from pathlib import Path
+
+import numpy as np
+
+from pydicom.dataset import Dataset, FileMetaDataset
+from pydicom.sequence import Sequence
+
+
+def write_volume(filename_pattern: str):
+    """Write a volumes worth of fake DICOM images
+
+    Args:
+        filename_pattern: The pattern to use for the filenames. This should include a {slice} which will be replaced with the slice number e.g. /tmp/slice{slice:03d}.dcm
+    """
+    # dicom_variables.json contains per slice information for a 3D image (geometry, windowing, etc.)
+    variables = json.load(open(Path(__file__).parents[2] / "resources" / "dicom_variables.json"))
+    for i, slice_info in enumerate(variables):
+        write_slice(
+            file_name=filename_pattern.format(slice=i),
+            pixel_data=np.random.rand(256, 256),
+            **slice_info
+        )
+
+
+def write_slice(
+    instance_creation_time: str,
+    sop_instance_uid: str,
+    instance_number: str,
+    image_position_patient: list[float],
+    slice_location: float,
+    window_centre: str,
+    window_width: str,
+    pixel_data: np.ndarray,
+    file_name: str,
+):
+    """Write a single fake DICOM image
+
+    Elements that vary between slices are exposed as arguments.  Values for these can be obtained from the dicom_variables.json file.
```

Comment:
I think it might help readability and reusability of this function if the constant data elements are loaded from elsewhere (e.g. another JSON file).

It's probably not needed for this PR, but if we're expecting to generate mock DICOM data often, maybe we could use a factory pattern (e.g. something like how it's done in https://github.com/sjoerdk/dicomgenerator)

Reply:
Thanks, I wasn't aware of dicomgenerator.  We're not doing much testing of the DICOM anonymisation process yet so I expect we'll need to be generating more test data.  Better to figure out our testing needs rather than trying to anticipate in this PR so not making further changes.